app/javascript/App/bundles/common/containers/NewsfeedContainer.tsx

```diff
@@ -31,43 +32,76 @@ class NewsfeedContainer extends React.Component<Props, State> {
     return Date.parse(y.feed_item_published_at) - Date.parse(x.feed_item_published_at)
   }
 
-  fetchAllNewsItems = () => {
-    this.setState({
-      status: STATUSES.LOADING,
-    },() => {
-      localAPI.get('/news').then((response) => {
-        this.setState({
-          status: STATUSES.READY,
-          sortedNewsItems: response.payload.sort(this.sortNewsFunc),
+  uniqNews = (arr) => {
+    return _.uniqBy(arr, elem => elem.id);
+  }
+
+  fetchNewNewsItems = (): Promise<NewsItem[]> => {
+    if (this.state.sortedNewsItems.length === 0) return Promise.resolve([]);
+
+    const firstNewsItem = this.state.sortedNewsItems[0];
+
+    return new Promise((resolve, _reject) => {
+      this.setState({
+        status: STATUSES.NEW_NEWS_ITEMS_LOADING
+      }, () => {
+        localAPI.get('/news', { publishedSince: firstNewsItem.feed_item_published_at }).then((response) => {
+          const existingNewsIds = this.state.sortedNewsItems.map(elem => elem.id);
+          const newNews = response.payload.filter((newsItem: NewsItem) => existingNewsIds.indexOf(newsItem.id) === -1);
```

Comment:
doesn't seem like your checking for the index, just whether the id is included. in that case it would be more elegant to use the `Array.includes` 

```
existingNewsIds.includes(newsItem.id)
```

Reply:
The reason was in that the typescript complained about `includes` function. Changed lib to `es7`